src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/client/AddressServiceClientServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.client;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import javax.inject.Inject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.validation.annotation.Validated;
+import uk.gov.ons.ctp.common.rest.RestClient;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressQueryRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.PostcodeQueryRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexSearchResultsDTO;
+
+/** This class is responsible for communications with the Addresse Index service. */
+@Service
+@Validated()
+public class AddressServiceClientServiceImpl {
+  private static final Logger log = LoggerFactory.getLogger(AddressServiceClientServiceImpl.class);
+
+  @Autowired private AppConfig appConfig;
+
+  @Inject
+  @Qualifier("addressIndexClient")
+  private RestClient addressIndexClient;
+
+  public AddressIndexSearchResultsDTO addressQuery(AddressQueryRequestDTO addressQueryRequest) {
```

Comment:
would prefer 'getAddresses' or 'findAddresses'

Reply:
I've gone for searchByAddress().

The naming convention works better for the postcode case.